packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -1188,4 +1189,105 @@ describe('Testing communication', () => {
       });
     });
   });
+  describe('validateOrigin', () => {
+    it("validateOrigin returns false if the protocol of origin is not 'https:'", () => {
+      /* eslint-disable-next-line @microsoft/sdl/no-insecure-url */ /* Intentionally using http here because of what it is testing */
+      const messageOrigin = new URL('http://teams.microsoft.com');
+      const result = validateOrigin(messageOrigin);
+      expect(result).toBe(false);
+    });
+    it('validateOrigin returns true if origin is in teams pre-known allowlist', () => {
+      const messageOrigin = new URL('https://teams.microsoft.com');
```

Comment:
>('https://teams.microsoft.com'); [](http://example.com/codeflow?start=35&length=32)

This unit test (and the ones like it) are assuming that teams.microsoft.com is in the validOrigins list. That might be true today, but it may not be true tomorrow.

I know this might seem overly picky, but as someone who recently had to update a bunch of unit tests that were making assumptions about which capabilities worked on which platform, we should try to minimize the assumptions the unit tests make so that they continue to run and pass even when unrelated changes are made. Even in this very change, you need to update unit tests that were making assumptions about message ordering, even though your change is not related to authentication or pages, etc.

In this case, rather hardcoding teams.microsoft.com, could you just pull the first element out of the `validOrigins` list and use that? That way we are always using one that we know is valid and even if that list changes, the test should still work.

Reply:
I really like this idea, I'll make that change now